### PR TITLE
Python/fix/raster tests

### PR DESCRIPTION
**Pull request by sllynn**
adding tests for raster functions to python bindings

### CODE REVIEW

File: python/mosaic/api/functions.py [hunk 1]
```diff
@@ -208,7 +208,9 @@ def st_bufferloop(
     )
 
 
-def st_buffer_cap_style(geom: ColumnOrName, radius: ColumnOrName, cap_style: ColumnOrName) -> Column:
+def st_buffer_cap_style(
```
Comment (milos-colic) [inline python/mosaic/api/functions.py:211]: we should remove this - it does nothing.
the JTS implementation as well, basically does the same normal buffer does :/

Comment (sllynn) [reply python/mosaic/api/functions.py]: OK, I've pulled it out. 👍

File: .github/actions/python_build/action.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ runs:
       shell: bash
       run: |
         cd python
-        pip install build wheel pyspark==${{ matrix.spark }}
+        pip install build wheel pyspark==${{ matrix.spark }} numpy==${{ matrix.numpy }}
```
Comment (sllynn) [inline .github/actions/python_build/action.yml:14]: When we move up to Ubuntu 22.04 / GDAL 3.6.x we can pull this back out.

**Review by milos-colic — APPROVED** [review]
LGTM!

File: src/main/resources/scripts/install-gdal-databricks.sh [hunk 3]
```diff
@@ -21,8 +21,8 @@ GDAL_RESOURCE_DIR=$(find $DATABRICKS_ROOT_VIRTUALENV_ENV -name "databricks-mosai
 
 # -- untar files to root
 # - from databricks-mosaic-gdal install dir
-tar -xf $GDAL_RESOURCE_DIR/resources/gdal-3.4.3-filetree.tar.xz -C /
+tar -xf $GDAL_RESOURCE_DIR/resources/gdal-3.4.3-filetree.tar.xz --skip-old-files -C /
```
[the remark this answers is not in the document]
Comment (sllynn) [reply src/main/resources/scripts/install-gdal-databricks.sh:24]: Running the script successfully is now part of the python test suite. The untar process is not idempotent and untar will raise an error if the target file(s) already exists. That's fine when running the script for the first time (on cluster start or in GH actions) but for local development you need to be able to re-run this multiple times without it hitting an error.

File: src/main/scala/com/databricks/labs/mosaic/core/raster/operator/CombineAVG.scala [hunk 4]
```diff
@@ -19,20 +19,18 @@ object CombineAVG {
...
-                           |    y = y / div
-                           |
-                           |    np.clip(y,0, sys.float_info.max, out = out_ar)
+                           |def average(in_ar, out_ar, xoff, yoff, xsize, ysize, raster_xsize, raster_ysize, buf_radius, gt, **kwargs):
+                           |    stacked_array = np.array(in_ar)
+                           |    pixel_sum = np.sum(stacked_array, axis=0)
+                           |    div = np.sum(stacked_array > 0, axis=0)
+                           |    div = np.where(div==0, 1, div)
+                           |    np.divide(pixel_sum, div, out=out_ar, casting='unsafe')
+                           |    np.clip(out_ar, stacked_array.min(), stacked_array.max(), out=out_ar)
```
[the remark this answers is not in the document]
Comment (sllynn) [reply src/main/scala/com/databricks/labs/mosaic/core/raster/operator/CombineAVG.scala:33]: Took forever to get it working using trial and error. Reading the docs helped a bit, mind you 🙄  Might be worth us building a little function that will allow us to pass in a script and inspect the internal state while it's executed?

**Pull request merged**